ethergo/client/client.go

```diff
@@ -96,13 +98,41 @@ func (c *clientImpl) getW3Client() *w3.Client {
 
 // BatchWithContext batches multiple w3 calls.
 func (c *clientImpl) BatchWithContext(ctx context.Context, calls ...w3types.Caller) (err error) {
+	// Do not create an even if there are no calls
+	if len(calls) == 0 {
+		return nil
+	}
+
 	ctx, span := c.tracing.Tracer().Start(ctx, batchAttribute)
 	span.SetAttributes(parseCalls(calls))
 	span.SetAttributes(attribute.String(endpointAttribute, c.endpoint))
 
 	defer func() {
+		if errors.Is(err, w3.CallErrors{}) {
+			var batchErr w3.CallErrors
+			_ = errors.As(err, &batchErr)
+			for i, callErr := range batchErr {
+				rawReq, err := calls[i].CreateRequest()
+				// this already happened, so it can't be failing now.
+				// just error.
+				if err != nil {
+					fmt.Println("could not create request: this should never happen", err)
```

Comment:
Should this be span event instead of a print? I guess doesn't really matter since it should never be hit?